xds/src/main/java/io/grpc/xds/EnvoyServerProtoData.java

```diff
@@ -150,9 +150,9 @@ abstract static class CidrRange {
 
     abstract int prefixLen();
 
-    static CidrRange create(String addressPrefix, int prefixLen) throws UnknownHostException {
+    static CidrRange create(String addressPrefix, int prefixLen) throws IllegalArgumentException {
```

Comment:
IllegalArgumentException is runtime, no need to specify it explicitly

Reply:
I think the takeaway should be "don't call InetAddresses.forString() here", not "remove the throws." We should pass in an `InetAddress addressPrefix` and do the `InetAddresses.forString()` in the caller.

Reply:
I respectfully disagree here. These factory methods are for our convenience, aren't they? They're POJO for protos we track, it's even in the class name - `EnvoyServerProtoData`.  Now that takes `InetAddress`, there's a chance someone will pass an object made as `InetAddress.getByName` instead of safe `InetAddresses.forString`. 
I believe the input should be a String, just like in proto, and then do parse it safely using `InetAddresses.forString`.

Reply:
`InetAddress.getByName()` is actually 100% okay... for IP addresses. It doesn't use DNS for IP addresses. The problem is _what if it isn't an IP address_. That is an equal problem when using `InetAddress.forString()`.

Any usage of `InetAddress.forString()` must handle malformed input. Unchecked exception is asking for trouble and we are receiving this data from a remote source, so we need to validate it. And if the caller must be responsible already for handling invalid cases, it isn't that big of a deal to have them do more of the parsing. After all, xds could have encoded the IP address as `bytes` if they so desired.

We have a _lot_ of APIs that use InetAddress. There's little special about this one to hide the InetAddress.